website_event_compassion/models/event_email.py

```diff
@@ -136,15 +123,27 @@ def _compute_scheduled_date(self):
     @api.multi
     def execute(self):
         """ Replace execute method to send communication instead of using
-        email template. """
+        email template.
+         return bool: true if all event_mail_registration were sent
+         """
+        sent_to_everyone = True
         for email in self:
-            registration = email.registration_id
-            if registration.state in ["open", "done"] and not email.mail_sent:
-                self.env["partner.communication.job"].create(
+            if email.mail_sent:
+                continue
+            try:
+                # if the communication fail by either an uncaught exception or is in the state failure
+                # skip this mail and dont mark it as sent
+                communication = self.env["partner.communication.job"].create(
                     {
-                        "partner_id": registration.partner_id.id,
-                        "object_ids": registration.ids,
+                        "partner_id": email.registration_id.partner_id.id,
+                        "object_ids": email.registration_id.ids,
                         "config_id": email.scheduler_id.communication_id.id,
                     }
                 )
+                if communication is None or communication.state in ["failure"]:
+                    raise Exception
```

Comment:
Why not putting the sent_to_everyone here instead of raising an exception catched just after?
```suggestion
                    sent_to_everyone = False
                    continue
```

Reply:
This cover also the case where the create communication also raise an exception

Reply:
I think the create method in communications already handle exceptions. I verified and when generating the text it catches them and puts the communication in failure state without raising the error.

I just saw one missing try/except when the number of pages is computed in method `def count_pdf_page`. One can be added there and here in the code I think we can remove this try/except clause.